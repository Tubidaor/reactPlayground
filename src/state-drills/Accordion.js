import React from 'react';


class Accordion extends React.Component {
  constructor(props){
    super(props)
  }

  static defaultProps = {
    sections: []
  }

  state = {
    currentIndex: 0
  }


  renderButtons = () => {
    return this.props.sections.map((section,index) => (
      <li>
        <button key={index} onClick={() => this.clickButton(index)}>
          {section.title}
        </button>
        {(index === this.state.currentIndex) && <p>{section.content}</p>}
      </li>
    ))
  }

  clickButton = (index) => {
    this.setState({
      currentIndex: index
    })
  }


  render() {

      return ( 
        <ul className='Accordion'>
          {this.renderButtons()}
        </ul>
      )
  }
}

export default Accordion;