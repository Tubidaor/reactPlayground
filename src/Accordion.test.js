import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './state-drills/Accordion.js'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { isMainThread } from 'worker_threads';
import { join } from 'path';

describe('Accordion Component', () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ] 

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div)
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the first section by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('closes the first section and opens new one', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})