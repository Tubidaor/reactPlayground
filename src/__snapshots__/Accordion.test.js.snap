// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first section and opens new one 1`] = `
<ul
  className="Accordion"
>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no tabs 1`] = `
<ul
  className="Accordion"
/>
`;

exports[`Accordion Component renders the first section by default 1`] = `
<ul
  className="Accordion"
>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
